### PR TITLE
added dll Kernel.EFSupport

### PR DESCRIPTION
**Pull request by vorobevamv**
added dll Kernel.EFSupport (+nuspec+yml);
IBaseDataProvider moved from Kernel/Database to Kernel.EFSupport/Provider
http://outsourcing.nat.tepkom.ru:20080/redmine/issues/1048

### CODE REVIEW

File: LT.DigitalOffice.Kernel.EFSupport.nuspec [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    <requireLicenseAcceptance>false</requireLicenseAcceptance>
+    <description>Library used in the Digital Office project.</description>
+    <releaseNotes>
+      Previous versions:
+    </releaseNotes>
+    <copyright>Copyright © LT Digital Office</copyright>
+    <tags>DigitalOffice</tags>
+    <dependencies>
+      <group targetFramework="net6.0">
+        <dependency id="SixLabors.ImageSharp" version="1.0.4"/>
```
Comment (MariyaSin) [inline LT.DigitalOffice.Kernel.EFSupport.nuspec]: ?

File: LT.DigitalOffice.Kernel.EFSupport.nuspec [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        <dependency id="LT.DigitalOffice.Kernel" version="1.1.59"/>
+        <dependency id="Svg" version="3.3.0"/>
```
Comment (MariyaSin) [inline LT.DigitalOffice.Kernel.EFSupport.nuspec]: ?

File: src/Kernel.EFSupport/Extensions/DatabaseExtension.cs [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+namespace LT.DigitalOffice.Kernel.EFSupport.Extensions
+{
+  public static class DatabaseExtension
+  {
+    public static void UpdateDatabase<TDbContext>
+      (this IApplicationBuilder app)
+      where TDbContext : DbContext
+    {
+      using IServiceScope serviceScope = app.ApplicationServices
+          .GetRequiredService<IServiceScopeFactory>()
```
Comment (MariyaSin) [inline src/Kernel.EFSupport/Extensions/DatabaseExtension.cs]: fix tab

File: src/Kernel.EFSupport/Extensions/DatabaseExtension.cs [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+          .CreateScope();
```
Comment (MariyaSin) [inline src/Kernel.EFSupport/Extensions/DatabaseExtension.cs]: fix tab

File: src/Kernel.EFSupport/Helpers/ConnectionStringExtension.cs [hunk 5]
```diff
@@ -0,0 +1,30 @@
+﻿using System;
+using LT.DigitalOffice.Kernel.Helpers;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Serilog;
+
+namespace LT.DigitalOffice.Kernel.EFSupport.Helpers
+{
+  public static class ConnectionStringExtension
```
Comment (MariyaSin) [inline src/Kernel.EFSupport/Helpers/ConnectionStringExtension.cs]: it is not connection string extension
rename to ConnectionStringHandler

File: src/Kernel.EFSupport/Helpers/ConnectionStringExtension.cs [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+  {
+    public static string AddConnectionString(this IServiceCollection services, IConfiguration _configuration)
```
Comment (MariyaSin) [inline src/Kernel.EFSupport/Helpers/ConnectionStringExtension.cs]: rename to GetConnectionString
what for   IServiceCollection here?

File: src/Kernel.EFSupport/Kernel.EFSupport.csproj [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+        <OutputType>Library</OutputType>
+    </PropertyGroup>
+    
+    <ItemGroup>
+        <PackageReference Include="LT.DigitalOffice.Kernel" Version="1.1.59" />
+        <PackageReference Include="Microsoft.EntityFrameworkCore" Version="6.0.5" />
+        <PackageReference Include="Microsoft.EntityFrameworkCore.InMemory" Version="6.0.5" />
+        <PackageReference Include="Microsoft.EntityFrameworkCore.Relational" Version="6.0.5" />
+        <PackageReference Include="Microsoft.EntityFrameworkCore.SqlServer" Version="6.0.5" />
+        <PackageReference Include="Serilog.AspNetCore" Version="4.1.0" />
```
Comment (MariyaSin) [inline src/Kernel.EFSupport/Kernel.EFSupport.csproj]: why serilog here?

File: src/Kernel.EFSupport/Extensions/DatabaseExtension.cs (hunk 3 above)
Comment (vorobevamv) [reply src/Kernel.EFSupport/Extensions/DatabaseExtension.cs]: done

File: src/Kernel.EFSupport/Extensions/DatabaseExtension.cs (hunk 4 above)
Comment (vorobevamv) [reply src/Kernel.EFSupport/Extensions/DatabaseExtension.cs]: done

File: src/Kernel.EFSupport/Helpers/ConnectionStringExtension.cs (hunk 5 above)
Comment (vorobevamv) [reply src/Kernel.EFSupport/Helpers/ConnectionStringExtension.cs]: OK

File: src/Kernel.EFSupport/Helpers/ConnectionStringExtension.cs (hunk 6 above)
Comment (vorobevamv) [reply src/Kernel.EFSupport/Helpers/ConnectionStringExtension.cs]: "GetConnectionString" - OK;
"this IServiceCollection services" for using this like extension in ConfigureServices (services.GetConnectionString);
don't we need this?

File: src/Kernel.EFSupport/Kernel.EFSupport.csproj (hunk 7 above)
Comment (vorobevamv) [reply src/Kernel.EFSupport/Kernel.EFSupport.csproj]: for "Log" and "using Serilog" in "GetConnectionString"

File: LT.DigitalOffice.Kernel.EFSupport.nuspec (hunk 1 above)
Comment (vorobevamv) [reply LT.DigitalOffice.Kernel.EFSupport.nuspec]: deleted

File: LT.DigitalOffice.Kernel.EFSupport.nuspec (hunk 2 above)
Comment (vorobevamv) [reply LT.DigitalOffice.Kernel.EFSupport.nuspec]: deleted

File: src/Kernel.EFSupport/Helpers/ConnectionStringExtension.cs (hunk 6 above)
Comment (vorobevamv) [reply src/Kernel.EFSupport/Helpers/ConnectionStringExtension.cs]: IServiceCollection deleted

File: src/Kernel.EFSupport/Kernel.EFSupport.csproj (hunk 7 above)
Comment (vorobevamv) [reply src/Kernel.EFSupport/Kernel.EFSupport.csproj]: fixed with new Kernel version (1.1.61)

File: src/Kernel.EFSupport/Helpers/ConnectionStringHandler.cs [hunk 8]
```diff
@@ -0,0 +1,29 @@
+﻿using System;
+using LT.DigitalOffice.Kernel.Helpers;
+using Microsoft.Extensions.Configuration;
+using Serilog;
+
+namespace LT.DigitalOffice.Kernel.EFSupport.Helpers
+{
+  public static class ConnectionStringHandler
+  {
+    public static string GetConnectionString(IConfiguration configuration)
```
Comment (MariyaSin) [inline src/Kernel.EFSupport/Helpers/ConnectionStringHandler.cs]: rename to "Get"

**Review by MariyaSin — APPROVED** [review]

File: src/Kernel.EFSupport/Helpers/ConnectionStringHandler.cs (hunk 8 above)
Comment (vorobevamv) [reply src/Kernel.EFSupport/Helpers/ConnectionStringHandler.cs]: renamed

**Review by VLiamin — APPROVED** [review]

**Pull request merged**